### PR TITLE
Move `WorkflowRunner` to `workflow` package

### PR DESCRIPTION
**Pull request by ellismg**
This refactoring moves WorkflowRunner into `workflow` (so it is now called `workflow.Runner`). This required breaking a direct dependency on the cobra and middleware packages, which was made possible by the newly introduced `workflow.AzdCommandRunner` interface which represents a component that can run commands like the `azd` CLI.

### CODE REVIEW

**Comment by ellismg** [comment]
Pulling this out of my vs-server refactoring - even if I end up not needing it for the vs-server work, I expect this is what you wanted all along, @wbreza.  Sorry for not having this insight earlier.

File: cli/azd/cmd/container.go [hunk 1]
```diff
@@ -596,10 +597,33 @@ func registerCommonDependencies(container *ioc.NestedContainer) {
 		container.MustRegisterNamedSingleton(platformName, constructor)
 	}
 
-	container.MustRegisterSingleton(NewWorkflowRunner)
+	container.MustRegisterSingleton(func(s ioc.ServiceLocator) (workflow.AzdCommandRunner, error) {
```
Comment (ellismg) [inline cli/azd/cmd/container.go:600]: This is sort of verbose, but I wasn't sure if there was a better way to request a named import here, so I have to do the indirect via the Service Locator.  I wish I could just instead inject a *cobra.Command and somehow denote the I wanted the one with the name `root-cmd`)

Comment (ellismg) [reply cli/azd/cmd/container.go:600]: Let me know if you have a better idea how to structure this, @wbreza.

**Review by wbreza — APPROVED** [review]
This was always my intent to have the workflow runner in the `workflow` package but got hung up on some circular dependency that you were able to fix via the interface which is great!

Thanks for doing this :)

File: cli/azd/cmd/container.go (hunk 1 above)
Comment (wbreza) [reply cli/azd/cmd/container.go:600]: Named singletons are tough to inject via constructor injection.  This is the same model I have used in some other components.  LGTM.

**Pull request merged**